presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1948,6 +1956,18 @@ public void revokeTablePrivileges(ConnectorSession session, SchemaTableName sche
         return grantInfos.build();
     }
 
+    @Override
+    public boolean isWritingStagingFilesEnabled(ConnectorSession session)
+    {
+        return isWritingStagingFileEnabled(session);
+    }
+
+    @Override
+    public void commitStagingFiles(ConnectorSession session, SchemaTableName tableName, Collection<Slice> fragments)
+    {
+        stagingFileCommitter.commitFiles(session, tableName, getPartitionUpdates(fragments));
```

Comment:
It looks like it can be done as the very first operation in the `finishInsert`/`finishCreateTable`. Is there any particular reason why is it added as a separate SPI?